src/Microsoft.AspNetCore.Mvc.Core/ProducesResponseTypeAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// Specifies the type of the value and status code returned by the action.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class ProducesResponseTypeAttribute : Attribute, IApiResponseMetadataProvider, IFilterMetadata
+    {
+        /// <summary>
+        /// Initializes an instance of <see cref="ProducesResponseTypeAttribute"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/> of object that is going to be written in the response.</param>
+        /// <param name="statusCode">Http response status code</param>
+        public ProducesResponseTypeAttribute(Type type, int statusCode)
```

Comment:
hahaha, I was about to complain about the name, if it's not too late why don't we just use Returns, ReturnsType, Result, ResultType or ProducesResult?

```
[Produces("application/json")]
[ProducesResponseType(typeof(Product))]
```

```
[Produces("application/json")]
[Returns(typeof(Product))]
```

```
[Produces("application/json")]
[ReturnsType(typeof(Product))]
```

@kichalla @Eilon @rynowak